### PR TITLE
Adds save for saveNewFederationBtcUTXOs test to FederationStorageProv…

### PR DESCRIPTION
**Pull request by jeremy-then**
Adds `save` `saveNewFederationBtcUTXOs` test.

### CODE REVIEW

**Review by apancorb — APPROVED** [review]
LGTM

**Review by wilmerrootstock — APPROVED** [review]
LGTM!

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java [hunk 1]
```diff
@@ -1085,6 +1085,45 @@ void setLastRetiredFederationP2SHScript() {
...
+        FederationStorageProvider federationStorageProvider = new FederationStorageProviderImpl(storageAccessor);
+
+        // Getting utxos from method
+        List<UTXO> actualUtxos = federationStorageProvider.getNewFederationBtcUTXOs(networkParameters, activations);
+
+        // Should be as the expected utxos
+        assertEquals(actualUtxos.size(), originalAmountOfUtxos);
+        assertEquals(expectedUtxos, actualUtxos);
+
+        List<UTXO> extraUtxos = BitcoinTestUtils.createUTXOs(1, btcAddress);
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:1109]: `extraUtxo` (singular) seems more accurate

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java [hunk 2]
```diff
@@ -1085,6 +1085,45 @@ void setLastRetiredFederationP2SHScript() {
...
+
+        actualUtxos.addAll(extraUtxos);
+
+        federationStorageProvider.save(networkParameters, activations);
+
+        int finalExpectedAmountOfUtxos = originalAmountOfUtxos + 1;
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:1115]: `originalAmountOfUtxos + extraUtxos.size()` ?

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java [hunk 3]
```diff
@@ -1085,6 +1085,45 @@ void setLastRetiredFederationP2SHScript() {
...
+        storageAccessor.saveToRepository(federationStorageIndexKey.getKey(), BridgeSerializationUtils.serializeUTXOList(expectedUtxos));
+
+        FederationStorageProvider federationStorageProvider = new FederationStorageProviderImpl(storageAccessor);
+
+        // Getting utxos from method
+        List<UTXO> actualUtxos = federationStorageProvider.getNewFederationBtcUTXOs(networkParameters, activations);
+
+        // Should be as the expected utxos
+        assertEquals(actualUtxos.size(), originalAmountOfUtxos);
+        assertEquals(expectedUtxos, actualUtxos);
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:1107]: same comment as [here](https://github.com/rsksmart/rskj/pull/2552#discussion_r1669109926)

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java [hunk 4]
```diff
@@ -1085,6 +1085,45 @@ void setLastRetiredFederationP2SHScript() {
 
     }
 
+    @ParameterizedTest
+    @MethodSource("provideFederationBtcUTXOsTestArguments")
+    void save_saveNewFederationBtcUTXOs_utxosShouldBeSavedToStorage(FederationStorageIndexKey federationStorageIndexKey, NetworkParameters networkParameters, ActivationConfig.ForBlock activations) {
```
Comment (julia-zack) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:1090]: remove the `save_` prefix

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java (hunk 3 above)
Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java]: Updated.

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java (hunk 4 above)
Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java]: Updated.

**Pull request closed without merging**